### PR TITLE
dns infoviz example

### PR DESCRIPTION
**Pull request by madestro**
by now covering:
- qtypes per IP per server
- queries per second

### CODE REVIEW

**Comment by madestro** [comment]
invalid merge

File: js/qtypes.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
+// qtypes d3 viz
+
+var url = 'http://localhost:8080/sse/queries_by_qtype';
```
Comment (franchoco) [inline js/qtypes.js]: Please put this line on a different file, so it is easily changeable by Docker. I suggest a file named "config.js".

Comment (franchoco) [inline js/qtypes.js]: I will do what I said in the previous commentary.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    align: left;
+    border: 1px;
+    border-style: solid;
+    border-radius: 8px;
+    padding: 5px;
+    }
+</style>
+</head>
+<body>
+<div class=rataheader>
```
Comment (franchoco) [inline index.html]: HTML attributes MUST be quoted: class="rataheader".

File: index.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
+<!DOCTYPE html>
+<html>
```
Comment (franchoco) [inline index.html]: This tag should have the charset attribute, in order to set the encoding.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+<head>
+<script src="https://d3js.org/d3.v3.min.js" charset="utf-8"></script>
+<style> /* set the CSS */
+
```
Comment (franchoco) [inline index.html]: If its possible, it would be nice to have the CSS on an external file.

File: js/qps.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
+// queries per sec d3 viz
+
+var url = server_url + '/sse/server_data_v2';
```
Comment (franchoco) [inline js/qps.js]: All variables are global if used this way. They might lead to name crashes and therefore bug that are hard to track.
A possibility is to use an Object as namespace, or anonymous functions with immediate call to not populate the global scope.

File: index.html (hunk 2 above)
Comment (madestro) [inline index.html]: done

File: js/qps.js (hunk 5 above)
Comment (madestro) [inline js/qps.js]: or all infoviz could be a function, so config.js will be the only file with global variables

File: index.html (hunk 2 above)
Comment (fmontoto) [inline index.html]: Do it for the entire code.

File: js/qtypes.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
+// qtypes d3 viz
+
+var url = server_url + '/sse/queries_by_qtype';
```
Comment (fmontoto) [inline js/qtypes.js]: This looks like something that must go to the config file.

File: js/qps.js (hunk 5 above)
Comment (madestro) [inline js/qps.js]: ok, i'll do that

File: index.html (hunk 3 above)
Comment (fmontoto) [inline index.html]: This isn't done @madestro

File: index.html (hunk 4 above)
Comment (fmontoto) [inline index.html]: Is this possible? @madestro

File: js/qtypes.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
+// qtypes d3 viz
+
+qtypes();
+
+function qtypes() {
+  var url = server_url + '/sse/queries_by_qtype';
```
Comment (fmontoto) [inline js/qtypes.js]: /sse/queries_by_qtype is a configuration parameter

File: index.html (hunk 3 above)
Comment (madestro) [inline index.html]: Doctypes have nothing to do with charsets: https://www.w3.org/wiki/Doctypes_and_markup_styles

File: index.html (hunk 4 above)
Comment (madestro) [inline index.html]: Done in https://github.com/niclabs/ratadns-remy/commit/0c86cf325158be21c6325d8a55fe62e372f4974f

File: js/qtypes.js (hunk 7 above)
Comment (madestro) [inline js/qtypes.js]: dunno, is hardly linked to the infoviz (the viz is for THAT service), should stay in that function IMHO

File: index.html (hunk 3 above)
Comment (fmontoto) [inline index.html]: Still missing the charset for the page.
Looks like you have to add <meta charset="utf-8">. See the same link or http://www.w3schools.com/html/html_charset.asp

Comment (madestro) [inline index.html]: Done in f51e4d7

File: js/qtypes.js (hunk 7 above)
Comment (fmontoto) [inline js/qtypes.js]: It's ok if it's hardly linked to infoviz, just refer to that link in the var name at config.
It's better to have all the configuration at one place. So when you have to change something you know exactly where to go.

**Pull request merged**